app/controller/source_data_reader.rb

```diff
@@ -1,4 +1,61 @@
 # frozen_string_literal: true
 
+require 'safe_yaml'
+require 'pry'
+
 class SourceDataReader
+
+  attr_writer :base_path
+
+  def initialize(base_path)
+    @base_path = base_path
+  end
+
+  def create_master_yaml
+    master = { 'topics' => topics, 'conf' => config }
+    File.open(@base_path + 'data/master.yaml', 'w') { |file| file.write(master.to_yaml) }
+  end
+
+  private
+
+  def topics
+    Dir.glob(@base_path + 'data/topics/*').map do |topic_path|
+      File.open(topic_path + '/topic.yaml') do |file|
+        id = topic_path[/[0-9]{1,3}$/]
```

Comment:
ID_REGEX = /[0-9]{1,3}$/.freeze

add constant since you're using it multiple times and it makes it more readable